.github/workflows/api-pull-request.yml

```diff
@@ -66,11 +68,31 @@ jobs:
           opts: --no-input --dry-run --check
         run: make django-make-migrations
 
+      - name: Restore cached testmon data
+        if: ${{ github.event_name != 'push' || github.ref != 'refs/heads/main' }}
+        id: cache-testmon-restore
+        uses: actions/cache/restore@v3
+        with:
+          enableCrossOsArchive: true
+          path: |
+            /home/runner/work/flagsmith/flagsmith/api/.testmondata*
+          key: testmon-data-python${{ matrix.python-version }}-${{ github.event.pull_request.base.sha }}
```

Comment:
`github.event.pull_request.base.sha` is not the same as `github.sha` used when caching the data, right? Is this intentional?

Reply:
Great, thanks for explaining. I'd love to see it explained in a comment because the reasons are not immediately obvious.

Reply:
I updated the PR description to mention those reasons as suggested.